packages/teuchos/parameterlist/src/Teuchos_ParameterList.cpp

```diff
@@ -457,6 +506,95 @@ void ParameterList::validateParameters(
 }
 
 
+void ParameterList::modifyParameterList(ParameterList & valid_pl,
+    int const depth)
+{
+  RCP<const ParameterListModifier> modifier;
+  if (nonnull(modifier = valid_pl.getModifier())) {
+    modifier->modify(*this, valid_pl);
+  }
+  ConstIterator itr;
+  for (itr = valid_pl.begin(); itr != valid_pl.end(); ++itr){
+    const std::string &entry_name = itr->first;
+    const ParameterEntry &entry = itr->second;
+    if (entry.isList() && depth > 0){
+      ParameterList &valid_pl_sublist = valid_pl.sublist(entry_name, true);
+      if(!valid_pl_sublist.disableRecursiveModification_){
+        const ParameterEntry *validEntry = this->getEntryPtr(entry_name);
+        TEUCHOS_TEST_FOR_EXCEPTION(
+          !validEntry, Exceptions::InvalidParameterName
+          ,"Error, the parameter {name=\""<<entry_name<<"\","
+          "type=\""<<entry.getAny(false).typeName()<<"\""
+          ",value=\""<<filterValueToString(entry)<<"\"}"
+          "\nin the parameter (sub)list \""<<this->name()<<"\""
+          "\nwas not found in the list of parameters during modification."
+          "\n\nThe parameters and types are:\n"
+          <<this->currentParametersString()
+          );
+        ParameterList &pl_sublist = this->sublist(entry_name, true);
+        pl_sublist.modifyParameterList(valid_pl_sublist, depth-1);
+      }
+    }
+  }
+}
+
+
+void ParameterList::reconcileParameterList(ParameterList & valid_pl,
+    const bool left_to_right){
```

Comment:
This is nit-picking but to match existing formatting can you push the `{` at the beginning of the next line?

Reply:
Fixed in next push